lasagne/init.py

```diff
@@ -18,12 +18,82 @@ def sample(self, shape):
 
 
 class Normal(Initializer):
-    def __init__(self, std=0.01, avg=0.0):
+    def __init__(self, std=0.01, mean=0.0):
         self.std = std
-        self.avg = avg
+        self.mean = mean
 
     def sample(self, shape):
-        return floatX(np.random.normal(self.avg, self.std, size=shape))
+        return floatX(np.random.normal(self.mean, self.std, size=shape))
+
+
+class Uniform(Initializer):
+    def __init__(self, range=0.01, std=None, mean=0.0):
+        import warnings
+        warnings.warn("The uniform initializer no longer uses Glorot et al.'s "
+                      "approach to determine the bounds, but defaults to the "
+                      "range (-0.01, 0.01) instead. Please use the new "
+                      "GlorotUniform initializer to get the old behavior. "
+                      "GlorotUniform is now the default for all layers.")
+
+        if std is not None:
+            a = mean - np.sqrt(3) * std
+            b = mean + np.sqrt(3) * std
+            self.range = (a, b)
+        elif isinstance(range, Number):
+            self.range = (-range, range)
+        else:
+            self.range = range
```

Comment:
That `isinstance` check looks a little brittle to me. For example, `isinstance(np.float32(10), Number)` returns `False`. What about:
```python
if std is not None:
    a = ...
    b = ...
else:
    try:
        a, b = range
    except (TypeError, ValueError):
        a, b = -range, range
self.range = (a, b)
```